lib/vquic/msh3.c

```diff
@@ -538,11 +677,44 @@ static void cf_msh3_destroy(struct Curl_cfilter *cf, struct Curl_easy *data)
   struct cf_msh3_ctx *ctx = cf->ctx;
 
   (void)data;
-  cf_msh3_ctx_clear(ctx);
+  cf_msh3_ctx_clear(cf, data);
   free(ctx);
   cf->ctx = NULL;
 }
 
+static CURLcode cf_msh3_query(struct Curl_cfilter *cf,
+                              struct Curl_easy *data,
+                              int query, int *pres1, void **pres2)
+{
+  struct cf_msh3_ctx *ctx = cf->ctx;
+
+  switch(query) {
+  case CF_QUERY_MAX_CONCURRENT: {
+    /* TODO: we do not have access to this so far, fake it */
```

Comment:
You're asking how many parallel requests the peer is currently allowing you to open? Why do you need that? msh3/msquic supports queueing requests until the peer is ready, so the app doesn't actually need to know how many instantaneous requests might be allowed (this value can change at any point in time).

Reply:
If the conditions are right, libcurl adds transfers to an existing connection if it can and multiplexes them. Otherwise it can create a new connection for it. For this to work, it needs to know if the new transfer can be put on the existing connection.